packages/typegpu/src/tgpuBindGroupLayout.ts

```diff
@@ -211,11 +236,23 @@ const DEFAULT_READONLY_VISIBILITY: TgpuShaderStage[] = [
   'fragment',
 ];
 
-function createBindGroupLayout<
-  Entries extends Record<string, TgpuLayoutEntry | null>,
->(entries: Entries): TgpuBindGroupLayout<Entries> {
-  return new TgpuBindGroupLayoutImpl(entries);
-}
+type InvokedRuntimeSized<
+  Entries extends Record<
+    string,
+    | TgpuLayoutEntry
+    | TgpuRuntimeSizedLayoutUniform
+    | TgpuRuntimeSizedLayoutStorage
+    | null
+  >,
+> = {
+  [K in keyof Entries]: Entries[K] extends TgpuRuntimeSizedLayoutUniform
+    ? TgpuLayoutUniform
```

Comment:
We are probably losing type information here. You can verify that by adding a test for the type of the bound entries. Maybe we can circumvent that if we do:

```suggestion
    ? Omit<Entries[K], 'uniform'> & { uniform: ReturnType<Entries[K]['uniform']> }
```

I would also experiment with merging the definition of TgpuLayoutUniform and TgpuRuntimeSizedLayoutUniform by just doing a union on:
```ts
uniform: AnyTgpuData | ((arrayLength: number) => AnyTgpuData);
```